packages/design-system/src/Dialog/index.tsx

```diff
@@ -61,12 +65,18 @@ export const DialogClient = ({
 			onKeyDown={handleKeyDown}
 			{...props}
 		>
-			<button type="button" onClick={handleClose} aria-label="Close gallery">
-				×
-			</button>
-			{children}
+			<div className={classes.container}>
+				<div className={classes.heading}>
+					<button type="button" onClick={handleClose} aria-label="Close dialog">
+						x
+					</button>
+					<h2 className={classes.title} aria-labelledby="modalTitle">
+						{title}
+					</h2>
+				</div>
```

Comment:
Prefer semantic elements when the role is clear

```suggestion
				<header className={classes.heading}>
					<button type="button" onClick={handleClose} aria-label="Close dialog">
						x
					</button>
					<h2 className={classes.title} aria-labelledby="modalTitle">
						{title}
					</h2>
				</header>
```